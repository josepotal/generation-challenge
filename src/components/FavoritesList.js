import React, {Component} from 'react';
import FavoriteItem from './FavoriteItem';

export default class FavoritesList extends Component {
  render () {
    const buttonText = (this.props.showFavs) ? 'Hide Favorite Store': 'Show Favorite Store'
    return(
      <div className="board">
        <h1>My Favorites Stores</h1>
        {(this.props.showFavs) ?
          this.props.favStores.map((store, index) => {
            return (
              <FavoriteItem
                key={index}
                store={store}
              />
            );
          })
        : <div>(Click on the icons to add them to your Favorites List)</div>
        }
        <button onClick={this.props.toggleFavsList}>{buttonText}</button>
      </div>
    )
  }
}
