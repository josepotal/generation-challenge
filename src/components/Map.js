import React, { Component } from 'react';
import PropTypes from 'prop-types';

import stores from '../utils/coords_store_directory.json'

const google = window.google;

export default class Map extends Component {
  constructor(){
    super()
    this.setMarkers = this.setMarkers.bind(this)
  }

  componentDidMount() {
    const {lat, lng} = this.props.initialPosition;
    this.map = new google.maps.Map(this.refs.map, {
      center: {lat, lng},
      zoom: 12
    });

    this.setMarkers(this.map)
  }

  setMarkers = map => {
    // eslint-disable-next-line
    stores.items.map((store, index) => {
      const image = {
        url: 'http://maps.gstatic.com/mapfiles/ms2/micons/blue.png',
        scaledSize: new google.maps.Size(42, 42),
        origin: new google.maps.Point(0, -6),
        anchor: new google.maps.Point(0, 0)
      }
      const imageFav = {
        url: 'http://maps.gstatic.com/mapfiles/ms2/micons/green.png',
        scaledSize: new google.maps.Size(42, 42),
        origin: new google.maps.Point(0, -6),
        anchor: new google.maps.Point(0, 0)
      }

      let marker;
      if (store.loc.length > 0){
        marker = new google.maps.Marker({
          position: {lat: store.loc[0], lng: store.loc[1]},
          map: map,
          animation: google.maps.Animation.DROP,
          title: store.Name,
          label: index.toString(),
          icon: image
        })

        //onClick add to my favorites list
        marker.addListener('click', () => {
          store = {
            ...store,
            id: index
          }
          this.props.handleAddFavs(store)
          marker.setIcon(imageFav)
        });

        var contentString = `
          <div class="content">
            <p>${index} | ${marker.title}</p>
            <p>${store.Address}</p>
          </div>`

        //onHover show info
        const infowindow = new google.maps.InfoWindow({
          content: contentString
        });

        marker.addListener('mouseover', function() {
          infowindow.open(map, marker);
        });
        marker.addListener('mouseout', function() {
          infowindow.close(map, marker);
        });
      }
    })
  }

  render() {
    return (
        <div className="mapContainer" ref="map"></div>
    );
  }
}

Map.propTypes = {
  initialPosition: PropTypes.object.isRequired,
  stores: PropTypes.array,
  store: PropTypes.object
};
