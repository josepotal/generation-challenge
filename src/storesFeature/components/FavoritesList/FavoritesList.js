import React from 'react';
import PropTypes from 'prop-types';

import FavoriteItem from '../FavoriteItem/FavoriteItem';
import styles from './FavoritesList.module.scss';

function FavoritesList({ showFavs, favStores, toggleFavsList }) {
  const buttonText = showFavs ? 'Hide Favorite Stores' : 'Show Favorite Stores';
  return (
    <div className={styles.favList}>
      <button className={styles.button} onClick={toggleFavsList}>
        {buttonText}
      </button>
      {showFavs ? (
        favStores.map(store => <FavoriteItem key={store.id} store={store} />)
      ) : (
        <div className={styles.message}>
          (Click on the icons to add them to your Favorites List)
        </div>
      )}
    </div>
  );
}

FavoritesList.propTypes = {
  showFavs: PropTypes.bool.isRequired,
  favStores: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      Name: PropTypes.string.isRequired,
      Address: PropTypes.string.isRequired,
      loc: PropTypes.arrayOf(PropTypes.number.isRequired),
    }),
  ).isRequired,
  toggleFavsList: PropTypes.func.isRequired,
};

export default FavoritesList;
