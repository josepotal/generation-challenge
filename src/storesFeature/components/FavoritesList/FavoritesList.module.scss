@value colors: '../../../styles/variables.scss';
@value blue from colors;

.favList {
  position: absolute;
  top: 70px;
  left: 70%;
  background-color: lighten(#ddd, 8%);
  width: 150px;
}

.button {
  width: 100%;
  height: 32px;
  font-size: 0.75em;
  font-weight: bold;
  display: block;
  background-color: white;
  color: #009be3;
  border: 2px solid #009be3;
  border-radius: 5px;
  cursor: pointer;
}

.message {
  font-size: 0.7em;
  color: blue;
  background-color: lighten(#ddd, 8%);
  text-align: center;
  border-radius: 5px;
}

//media query small screens
@media screen and (max-width: 640px) {
  .favList {
    position: absolute;
    top: 70px;
    left: 37%;
    background-color: lighten(#ddd, 8%);
    width: 150px;
  }
}
