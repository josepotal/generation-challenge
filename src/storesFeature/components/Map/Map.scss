@value colors: '../../../styles/variables.scss';
@value lightGrey from colors;

.mapContainer {
  margin: 0 auto;
  width: 100%;
  height: 100vh;
}

.content {
  max-width: 160px;
  color: lightGrey;
  text-align: center;
  p:nth-child(1) {
    font-size: 0.85em;
    font-weight: bold;
  }
  p:nth-child(2) {
    font-size: 0.7em;
  }
}
