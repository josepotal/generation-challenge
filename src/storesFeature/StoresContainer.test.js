import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';

import { createMockStore } from 'redux-test-utils';

import FavoritesList from './components/FavoritesList/FavoritesList';

import Map from './components/Map/Map';
import ConnectedStoresContainer, { StoresContainer } from './StoresContainer';

/* Test the connected component */
describe('StoresContainer connected', () => {
  const initialState = {
    favStores: [],
    showFavs: false,
  };
  const store = createMockStore(initialState);
  // const store = createStore(reducer);

  const connectedComponent = shallow(
    <Provider store={store}>
      <ConnectedStoresContainer />
    </Provider>,
  );
  /* Tests go here */
  it('always renders a div', () => {
    expect(connectedComponent.find(ConnectedStoresContainer).length).toBe(1);
  });
});

describe('StoresContainer not connected', () => {
  const actions = {
    addFavoriteStore: jest.fn(),
    toggleFavsList: jest.fn(),
  };

  const state = {
    favStores: [],
    showFavs: false,
  };
  const component = shallow(
    <StoresContainer
      showFavs={state.showFavs}
      favStores={state.favStores}
      actions={actions}
    />,
  );

  /* Tests go here */
  it('always renders a div', () => {
    expect(component.find('div').length).toBeGreaterThan(0);
  });

  it('it should render `Favorites List`', () => {
    expect(component.find(FavoritesList).length).toBe(1);
  });

  it('it should render `Map`', () => {
    expect(component.find(Map).length).toBe(1);
  });

  it('should have a handleAddFavs function', () => {
    expect(component.instance().handleAddFavs.length).toBe(1);
  });

  // it('should have a toggleFavsList function', () => {
  //   expect(component.instance().toggleFavsList.length).toBe(1);
  // });

  it('it should receive a store prop', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
