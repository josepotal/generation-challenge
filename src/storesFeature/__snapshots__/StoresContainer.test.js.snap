// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoresContainer not connected it should receive a store prop 1`] = `
<div>
  <Map
    favStores={Array []}
    handleAddFavs={[Function]}
    initialPosition={
      Object {
        "lat": 19.43236,
        "lng": -99.1332,
      }
    }
  />
  <FavoritesList
    favStores={Array []}
    showFavs={false}
    toggleFavsList={[Function]}
  />
</div>
`;
